spec/integration/lookup_spec.rb

```diff
@@ -94,13 +94,9 @@
 
     context 'with interpolations' do
       let(:hiera_config) { File.join(project, 'hiera_interpolations.yaml') }
-
-      it 'returns an error' do
+      it 'returns a value' do
         result = run_cli_json(cli_command + %w[key=test::interpolations])
-        expect(result).to include(
-          'kind' => 'bolt/pal-error',
-          'msg'  => /Interpolations are not supported in lookups/
-        )
+        expect(result).to eq('test::interpolations data/common.yaml')
```

Comment:
We want to ensure bolt errors here. Returning a success here is a regression.

Reply:
Sorry, I completely forgot about https://github.com/puppetlabs/bolt/pull/3213 I think you can update the test to validate that the default value is used in the lookup.

Reply:
is this similar to other tests were we add options with a default value to the run_cli_json call?

```
    it 'accepts a default value' do
      options = { 'default_value' => 'default' }.to_json
      result  = run_cli_json(cli_command + %W[key=foo::bar::baz options=#{options}])
      expect(result).to eq('default')
    end
```

Reply:
The core of the issue here is that we wanted to make it clear to users that in plan scope (specifically *not* in an apply block) there is not going to be the context (for example `facts`) necessary to resolve a hierachy that relies on that data for resolution. Previously we relied on a behavior in puppet whereby a lookup without necessary context would result in an error. Turns out that was not necessarily the correct assumption (see https://github.com/puppetlabs/bolt/pull/3213 ). The path forward here is that a lookup attempted in plan scope (outside of an apply block) should be able to do a lookup without error using the defaults that the underlying puppet/hiera comes up with. We will need to update our tests to codify our expectations about what this behavior should be. Generally, does the test code in these fixtures https://github.com/puppetlabs/bolt/tree/main/spec/fixtures/hiera make sense to you? The idea is that we have some plan code and a bunch of hiera configurations used in the tests to validate assumptions about what lookups will resolve to.

Reply:
would https://github.com/puppetlabs/bolt/blob/main/spec/fixtures/hiera/modules/test/data/common.yaml need updated for this test? I'm not very familar with hiera so i'm wondering if theres any similar tests you could point me to?

Reply:
The tests in this file largely cover the expectations for how lookups behave. The data itself in that file (`common.yaml`) should remain unchanged, the methods we use for doing the lookup (for example in top level plan scope) with the various configuration files we have there and the expected results will need to be refactored to ensure the defaults are expected. 

This section of our docs does a decent job of explaining the basics. https://www.puppet.com/docs/bolt/latest/hiera.html You can review that and make sure that the new behavior still tracks with the docs.